### DISCUSSION TITLE
Invalid code signing certificate

### DISCUSSION THREAD

**Discussion by Thumbler1983** [codemagic-ci-cd/codemagic-docs#2138] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

Launch build on dev branch

### Expected results

Build completes successfully

### Actual results

Build fails during "Set up code signing identities" step with the following error:

**Build failed :|
Invalid code signing certificate**

Nothing has been changed in the configuration before the builds start failing. It suddenly happened the 16th of October.
I've already tried to generate a brand new certificate and provisioning files but it did not work.

### Build id (optional)

653622d2d14fb8ea05ad58b3

**Comment by dtrdic** [comment, marked as answer]
Hi @Thumbler1983, I noticed that you've specified the Xcode version as the `edge` version, which is currently Xcode 15.1. Please set it to Xcode 14.3.1, which was the version you were using when the build was successful

**Reply by Thumbler1983** [reply]
Thank you so much @dtrdic, it worked. But... it seems more or less like a workaround. I use Xcode 15 locally and ad hoc builds work just fine. Is there anything I can do in the future to solve this issue? For the time being your fix is good enough anyway. Thanks again.

**Reply by dtrdic** [reply]
I'm glad that helped!:) You can still use other Xcode versions, but the key takeaway here is that if your app is building successfully locally with a specific set of versions (Flutter, Xcode, CocoaPods, etc.), you should set the build environment in your CI/CD pipeline with those same versions, like this you will avoid compatibility issues
